config/config_mgmt.py

```diff
@@ -107,10 +110,146 @@ def readConfigDB(self):
         configdb.connect()
         deep_update(data, FormatConverter.db_to_output(configdb.get_config()))
         self.configdbJsonIn =  FormatConverter.to_serialized(data)
-        self.logInFile('Reading Input', self.configdbJsonIn, True)
+        #self.logInFile('Reading Input', self.configdbJsonIn, True)
+
+        return
+
+    def writeConfigDB(self, jDiff):
+        print('Writing in Config DB')
+        """
+        On Sonic Switch
+        """
+        db_kwargs = dict(); data = dict()
+        configdb = ConfigDBConnector(**db_kwargs)
+        configdb.connect(False)
+        deep_update(data, FormatConverter.to_deserialized(jDiff))
+        data = sort_data(data)
+        self.logInFile("Write in DB: Last Data\n", data)
+        configdb.mod_config(FormatConverter.output_to_db(data))
+
+        return
+
+    """
+      Check if a key exists in ASIC DB or not.
+    """
+    def checkKeyinAsicDB(key):
+
+        self.logInFile('Check Key in Asic DB: {}'.format(key))
+        try:
+            # chk key in ASIC DB
+            if oid and db.exists('ASIC_DB', key):
+                return True
+        except Exception as e:
+            print(e)
+            raise(e)
+
+        return False
 
+    def testRedisCli(key):
+        # To Debug
+        if self.DEBUG_FILE:
+            cmd = 'sudo redis-cli -n 1 hgetall '"{}"'.format(key)
+            self.logInFile("Running {}".format(cmd))
+            print(cmd)
+            system(cmd)
         return
 
+    """
+     Check ASIC DB for PORTs in port List
+     ports: List of ports
+     portMap: port to OID map.
+     Return: True, if all ports are not present.
+    """
+    def checkNoPortsInAsicDb(self, db, ports, portMap):
+        try:
+            # connect to ASIC DB,
+            db.connect(db.ASIC_DB)
+            for port in ports:
+                key = self.oidKey + portMap[port]
+                if checkKeyinAsicDB(key) == False:
+                    # Test again via redis-cli
+                    testRedisCli(key)
```

Comment:
Do we need this double check?

Reply:
With the verbose option, we are double-checking, because I feel this is the best way to test this code. i.e. with -v option, run test code as well.